src/Entity/Category.php

```diff
@@ -5,36 +5,66 @@
 use Doctrine\Common\Collections\ArrayCollection;
 use Doctrine\Common\Collections\Collection;
 use Doctrine\ORM\Mapping as ORM;
+use Symfony\Component\Validator\Constraints as Assert;
+use Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity;
 
 /**
  * Category
  *
  * @ORM\Table(name="Category")
- * @ORM\Entity
+ * @UniqueEntity("name")
+ * @ORM\Entity(repositoryClass="App\Repository\CategoryRepository")
  */
 class Category
 {
     /**
      * @var int
      *
-     * @ORM\Column(name="id", type="integer", nullable=false)
+     * @ORM\Column(name="id", type="integer", nullable=false, options={"unsigned"=true})
      * @ORM\Id
      * @ORM\GeneratedValue(strategy="IDENTITY")
      */
     private $id;
 
     /**
      * @var string
-     * @ORM\Column(name="name", type="string", length=255, nullable=false)
+     * @ORM\Column(name="name", type="string", length=50, nullable=false)
+     * @Assert\NotBlank(message = "This value cannot be blank.")
```

Comment:
Nie jestem specjalnie fanem wrzucania walidatorów bezpośrednio w modelu. Narzucają one całemu systemu odpowiednią strukturę i nie są zbytnio elastyczne. Ja bym je wrzucił bardziej do samych formularzy, gdzie na podstawie danych przesłanych w POST można odpowiednio włączać / wyłączać walidatory oraz nimi sterować. Przykładowo, jeśli chcemy sprawdzić unikalność pola $name w danej grupie będzie to wymagać pobrania parametru parent z requesta oraz zbudowania odpowiedniego zapytania. Szczerze, nie wiem czy da się to prosto osiągnąć korzystając z wbudowanych klas symfony ;)